docs/reference/mapping/types/histogram.asciidoc

```diff
@@ -0,0 +1,87 @@
+[role="xpack"]
+[testenv="basic"]
+[[histogram]]
+=== Histogram datatype
+++++
+<titleabbrev>Histogram</titleabbrev>
+++++
+
+A `histogram` field stores pre-aggregated numerical data represented as a histogram. These histograms
+are defined by a set of pairs, one being the value representing the bucket and the count of that value.
+THe values should be defined accordingly to the type of aggregation this field will be used for.
+
+Histograms can be used for aggregations, in particular <<search-aggregations-metrics-percentile-aggregation, percentiles>>
+and <<search-aggregations-metrics-percentile-rank-aggregation, percentile ranks>> aggregations are supported.
+
+Histograms are indexed by defining two arrays, one containing the `values` (or buckets) for the histogram and the other containing
+the `counts` for such values. Those two arrays must have the same length.
+
```

Comment:
Not strictly needed for MVP, but it might be nice to add some text for context and split up the index creation and document indexing snippets.

For example:

```
[[histogram-ex]]
==== Examples

The following <<indices-create-index, create index>> API request creates a new index with two field mappings:

* `my_histogram`, a `histogram` field used to store percentile data
* `my_text`, a `keyword` field used to store a title for the histogram

[ INSERT CREATE INDEX SNIPPET ]
...

The following <<docs-index_,index>> API requests store pre-aggregated for two histograms: `histogram_`` and `histogram_2`.

[ INSERT DOC INDEX SNIPPET ]
...
```

Providing an example use case for the data may also be helpful. For example, the histograms could represent load time, similar to the percentile aggs docs. Not required for MVP though.